### ISSUE TITLE
test_axes:test_polar_unit broke unintentionally

### ISSUE THREAD

**Issue by mdboom** [matplotlib/matplotlib#5530]
In #5307, where I am lowering our testing tolerance down to zero, I am discovering a few recent changes that slipped through because the tolerance on our tests was too high.

The test broke somewhere in ebb2e63, ef683df2 or c326744 (not all of those commits are buildable).

BASELINE IMAGE:

![polar_units_2-expected](https://cloud.githubusercontent.com/assets/38294/11300511/cc2f2cb8-8f5d-11e5-88ab-b1dd595c0c1d.png)

CURRENT RESULT:

![polar_units_2](https://cloud.githubusercontent.com/assets/38294/11300518/d98920bc-8f5d-11e5-8c25-df535b0d5b31.png)

Test code:

```
    fig = plt.figure()

    # make sure runits and theta units work
    y1 = [y*km for y in y1]
    plt.polar(x2, y1, color="blue", thetaunits="rad", runits="km")
    assert_true(isinstance(plt.gca().get_xaxis().get_major_formatter(), units.UnitDblFormatter))
```

Cc: @cimarronm as the author of the bisect commits

**Comment by mdboom** [comment]
Maybe this was intentional?  The `km` has moved from the center to the outside.  I can certainly see how that's an improvement.

**Comment by dopplershift** [comment]
Well, other than the fact that it's directly underneath the tick label out there. I actually didn't notice it until you pointed it out--I thought `'km'` had disappeared.

**Comment by mdboom** [comment]
Closed by #5307.
